### PR TITLE
Add support for hours like 'Betriebsstundenzähler'

### PR DESCRIPTION
**Pull request by hacki11**
Supports Time-based counters with 'hours' as unit

- add Time category
- add Hours as unit
- fix smaller eslint issues

Feel free to review

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -179,6 +179,18 @@ class Sourceanalytix extends utils.Adapter {
             }
         }
 
+        // Load time definitions
+        catArray = ['Hour'];
+        for (const item in catArray) {
+            const unitItem = adapterHelpers.units.time[catArray[item]];
```
Comment (DutchmanNL) [inline main.js:185]: having the section loaded on .time limits us to only calculate hour. In current why of unit transition I cannot handle hour conversation (like s => m => h) as the exponents are missing.,

will check with helper lib adding the missing values and than we can use all time units including conversion

File: admin/custom_m.html [hunk 2]
```diff
@@ -155,6 +155,11 @@
                 { "unit": "cl", "exponent": -2 },
                 { "unit": "ml", "exponent": -3 }
             ]
+        },
+        "time": {
```
Comment (DutchmanNL) [inline admin/custom_m.html:159]: we should implement the complete lib

File: admin/index_m.html [hunk 3]
```diff
@@ -72,6 +72,11 @@
 					{ "unit": "cl", "exponent": -2 },
 					{ "unit": "ml", "exponent": -3 }
 				]
+			},
+			"time": {
```
Comment (DutchmanNL) [inline admin/index_m.html:76]: we should implement the complete lib
```
        "Second": [
            {"unit": "s", "exponent": 0}
        ],
        "Minute": [
            {"unit": "min", "exponent": 0}
        ],
        "Hour": [
            {"unit": "h", "exponent": 0}
        ],
        "Day": [
            {"unit": "d", "exponent": 0}
```

File: admin/index_m.html [hunk 4]
```diff
@@ -296,7 +301,7 @@ <h3>SourceAnalytix</h3>
 										<th data-name="unitType" class="translate"
 											style="width: 10%; text-align: center;" data-default="kWh"
 											data-type="select" data-style="text-align: center"
-											data-options="GW;MW;kW;W;mW;GWh;MWh;kWh;Wh;mWh;km³;m³;dm³;cm³;mm³;hl;l;dl;cl;ml">
+											data-options="GW;MW;kW;W;mW;GWh;MWh;kWh;Wh;mWh;km³;m³;dm³;cm³;mm³;hl;l;dl;cl;ml;h">
```
Comment (DutchmanNL) [inline admin/index_m.html:304]: we should implement the complete lib
```
s;min;h;d
```

**Review by DutchmanNL — CHANGES_REQUESTED** [review]

**Comment by DutchmanNL** [comment]
thank you, interesting idea please have a look at my comments

File: main.js (hunk 1 above)
Comment (hacki11) [reply main.js:185]: We can not use exponents here I think. Aren't they of base 10? Time calculations are 60 and 24. Perhaps a 'factor' with the real number would be more sufficient as a base 10 exp

Comment (DutchmanNL) [reply main.js:185]: good one, naming it exponents would incorrect its indeed a factor calculation. the factor should be set in the library we will only consume it basically in SA

Comment (hacki11) [reply main.js:185]: A really crappy hack would be log(60) and log(24). That would end in a correct base 10 exponent. But we will get problems with incorrect calculations due to rounding issues 🥳

**Pull request closed without merging**